Dockerfile

```diff
@@ -3,13 +3,13 @@ FROM jboss/base-jdk:8
 
 # Set the WILDFLY_VERSION env variable
 ENV WILDFLY_VERSION 9.0.0.Final
+ENV WILDFLY_SHA1 6a6c369581da1f1eb606512f144cb6aafa19a7ef
+ENV JBOSS_HOME /opt/jboss/wildfly
 
 # Add the WildFly distribution to /opt, and make wildfly the owner of the extracted tar content
 # Make sure the distribution is available from a well-known place
-RUN cd $HOME && curl http://download.jboss.org/wildfly/$WILDFLY_VERSION/wildfly-$WILDFLY_VERSION.tar.gz | tar zx && mv $HOME/wildfly-$WILDFLY_VERSION $HOME/wildfly
-
-# Set the JBOSS_HOME env variable
-ENV JBOSS_HOME /opt/jboss/wildfly
+RUN cd $HOME && curl -O https://download.jboss.org/wildfly/$WILDFLY_VERSION/wildfly-$WILDFLY_VERSION.tar.gz 
```

Comment:
This will leave the `tar.gz` in one layer making the result image heavier by more than 100MB. You should combine both RUN statements. And remove the tar.gz later.